### PR TITLE
chore: fixes for lean4#2688

### PR DESCRIPTION
**Pull request by semorrison**

### CODE REVIEW

**Comment by semorrison** [comment]
I've now merged this into `nightly-testing`.

It would be good to have approval on this before the next Lean release (next Sunday).

**Review by digama0 — APPROVED** [review]
Merge this when the upstream PR lands

**Review by joehendrix — APPROVED** [review]

**Pull request closed without merging**
